exp/addons/controllers/clusterresourceset_controller.go

```diff
@@ -211,6 +209,11 @@ func (r *ClusterResourceSetReconciler) ApplyClusterResourceSet(ctx context.Conte
 				conditions.MarkFalse(clusterResourceSet, addonsv1.ResourcesAppliedCondition, addonsv1.WrongSecretTypeReason, clusterv1.ConditionSeverityWarning, err.Error())
 			} else {
 				conditions.MarkFalse(clusterResourceSet, addonsv1.ResourcesAppliedCondition, addonsv1.RetrievingResourceFailedReason, clusterv1.ConditionSeverityWarning, err.Error())
+
+				// Continue without adding the error to the aggregate if we can't find the resource.
```

Comment:
I'm missing a lot of context here so sorry for the newbie question:

I would think the Secret/ConfigMap resource not existing would still result in an error and a retry, what's the reason for skipping it here?

Reply:
That would be tackled in a different PR, which will add watches on both ConfigMaps/Secrets we're interested in. When a new secret gets created, we get notified and we reconcile the linked CRS resources

Reply:
I see, even in that case, if during reconcile the resource got deleted and was no longer available shouldn't we err and retry?

Reply:
If resource gets deleted, no need to retry until it is recreated again and we will be watching create and update events on ConfigMaps/Secrets. Do you agree?

Reply:
I think so! Should we add this check when the ConfigMap/Secret watch is added? Or does this watch already exist? Maybe I missed it.

Reply:
That change is tracked here: https://github.com/kubernetes-sigs/cluster-api/issues/3397
Will open up a PR soon.

Reply:
gotcha, thanks!